File: 00-每日一题/20200216_80.cpp
//
// Created by zanbo on 2020/2/16.
//
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size()<=2)
            return nums.size();
        int length = 0;
        int flag = 0;
        for(int i=1;i<nums.size();i++)
        {
            if(nums[i]!=nums[i-1])
            {
                nums[++length] = nums[i];
                flag=0;
            }
            else if(nums[i]==nums[i-1] && flag==0)
            {
                nums[++length] = nums[i];
                flag++;
            }
        }
        return length+1;
    }
};

File: 00-每日一题/20200216_80.py
class Solution(object):
    def removeDuplicates(self, nums):
        length = len(nums)

        if length < 2:
            return length

        j = 1

        for i in range(2, length):
            if nums[i] != nums[j-1]:
                j = j+1
                nums[j] = nums[i]

        return j + 1
